solutions/JOI 18-commuter-pass.mdx

```diff
@@ -1,13 +1,21 @@
 ---
 id: joi-18-commuter-pass
 title: JOI 2018 Commuter Pass
-author: Andi Qu
+author: Andi Qu, Nathan Wang
 ---
 
-<IncompleteSection />
+1. Use Dijkstra's to calculate distances from $s$, $u$, and $v$.
+2. Consider the dijkstra tree (keep track of all the parents of each node) from $s$ to $t$. Note that this is a DAG.
+3. We basically need to choose one path on this DAG (any path on the DAG is a valid commuter pass option), a starting node for $u$, and an exiting node for $v$.
+4. Define `dpForward(i)` = min dist from $v$ to any node that lies on a path from $i$ _forwards_ through the DAG to $t$.
+5. Define `dpBackward(i)` = min dist from $v$ to any node that lies on a path from $i$ _backwards_ through the DAG to $s$; in other words, the min dist from $v$ to any node $x$ such that $x$ is in a path from $s$ to $i$ in the DAG.
```

Comment:
Maybe include what the recurrences are for these? It wasn't immediately obvious iirc

Reply:
oops we had slightly different DP definitions, I'll go with yours :)